.changeset/curvy-foxes-reply.md

```diff
@@ -0,0 +1,16 @@
+---
+'astro': minor
+---
+
+Move generated content collection types to a `.astro` directory. This replaces the previously generated `src/content/types.generated.d.ts` file.
+
+If you're using Git for version control, we recommend ignoring this generated directory by adding `.astro` to your .gitignore.
+
+#### Migration
+
+You will need a [TypeScript reference path](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-path-) to include `.astro` types in your project. Running `astro dev`, `astro build`, or `astro sync` will configure this automatically if your project has a `src/env.d.ts` file. Otherwise, you can add a `src/env.d.ts` file manually with the following contents:
+
+```diff
+/// <reference path="astro/client" />
++ /// <reference types="../.astro/types.d.ts" />
```

Comment:
Does this mean you don't get types without manually creating a file? Or was this already required? Is there anything we can do about that?

Reply:
This will be automatically generated if a `src/env.d.ts` is present! Otherwise, we assume you've deleted or moved this file to manage ambient types yourself. Since this file is home to the `astro/client` types, I'd assume most users have this file in their Astro project.